### PR TITLE
Add PHP 8.0 support

### LINKED ISSUE #13: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by bfoosness**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #13

### CODE REVIEW

File: test/Validator/AuthenticationTest.php [hunk 1]
```diff
@@ -211,12 +211,7 @@ public function testNoServiceThrowsRuntimeException()
 
     public function testEqualsMessageTemplates()
     {
-        $validator = $this->validator;
-        $this->assertAttributeEquals(
-            $validator->getOption('messageTemplates'),
-            'messageTemplates',
-            $validator
-        );
+        $this->assertGreaterThan(0, count($this->validator->getOption('messageTemplates')));
```
Comment (bfoosness) [inline test/Validator/AuthenticationTest.php]: I welcome guidance on this change. What does the previous test confirm, and does this change suffice?

Comment (svycka) [reply test/Validator/AuthenticationTest.php]: maybe like this?
```suggestion
        $r = new ReflectionObject($validator);
        $p = $r->getProperty('messageTemplates');
        $p->setAccessible(true);
        $this->assertEquals($validator->getOption('messageTemplates'), $p->getValue($validator));
```

**Comment by holtkamp** [comment]
@boesing this PR seems to be complete to satisfy https://github.com/laminas/laminas-authentication/issues/13, does it need additional effort/work?

**Review by Slamdunk — APPROVED** [review]

**Comment by bfoosness** [comment]
The laminas-db dependency is no longer a blocker, but I see I have a couple of issues to fix on this one now. I'll look at those this morning.

**Comment by weierophinney** [comment]
I've restored the test case to be equivalent to what it was previously (not that it's terribly useful), and rebased your branch off current 2.8.x (which adds our GHA workflow) - all looks good, so I'm merging!

**Pull request merged**